pkg/rtc/mediatrackreceiver.go

```diff
@@ -31,7 +32,7 @@ type MediaTrackReceiver struct {
 
 	lock            sync.RWMutex
 	receiver        sfu.TrackReceiver
-	layerDimensions sync.Map // livekit.VideoQuality => *livekit.VideoLayer
+	layerDimensions map[livekit.VideoQuality]*livekit.VideoLayer
```

Comment:
For some reason that I do not yet understand, this was reporting races a lot. So, making it a map and using locks.

Reply:
I wonder if this value is being copied. From the docs, `A Map must not be copied after first use.` - same as with the sync Mutexes. So although I can't see it in this diff, I wonder if this field is being copied out to a variable or something somewhere, which might be triggering the race dector (go vet I think should show it too if it is).

Reply:
Maybe. Will try to check that out.